src/MountedTraversal.js

```diff
@@ -50,6 +50,9 @@ export function instHasClassName(inst, className) {
     return false;
   }
   let classes = findDOMNode(inst).className || '';
+  if (typeof classes !== 'string') {
+    return false;
```

Comment:
@ljharb the string operations are the current implementation, I just added the `classList` check. We can definitely look at implementing the `className` branch differently though.